docs/tutorials/_static/geomesa-examples-jupyter/shallow-join-gdelt.ipynb

```diff
@@ -0,0 +1,814 @@
+    "                (featureA, featureB)\n",
+    "            } else {\n",
+    "                (featureB, featureA)\n",
+    "            }\n",
+    "        \n",
+    "        // Loop over the aggregate feature's properties, adding on the regular feature's properties\n",
+    "        aggFeature.getProperties.foreach{prop =>             \n",
+    "            val name = prop.getName.toString\n",
+    "            if (name.startsWith(\"total_\")) {\n",
+    "                val geoProp = geoFeature.getProperty(name.substring(6))\n",
+    "                if (geoProp != null) {\n",
+    "                    val sum = (prop.getValue, geoProp.getValue) match {\n",
+    "                        case (a: Integer, b: Integer) => a + b\n",
+    "                        case (a: java.lang.Long, b: java.lang.Long) => a + b\n",
+    "                        case (a: java.lang.Double, b: java.lang.Double) => a + b\n",
+    "                        case _ => 0\n",
+    "                    }          \n",
+    "                    aggFeature.setAttribute(name, sum)\n",
+    "                }\n",
+    "            }\n",
+    "\n",
+    "        }\n",
+    "        aggFeature.setAttribute(\"count\", aggFeature.get[Integer](\"count\") + 1)\n",
+    "        aggFeature\n",
+    "    } \n",
+    "    \n",
+    "    result\n",
+    "})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Send a map of country name -> geom to the executors"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "val countryMap: scala.collection.Map[String, Geometry] = \n",
+    "    reduced.map{ case (key, sf) => \n",
+    "        (sf.getAttribute(\"NAME\").asInstanceOf[String] -> sf.getAttribute(\"the_geom\").asInstanceOf[Geometry])\n",
+    "    }.collectAsMap\n",
+    "    \n",
+    "val broadcastedCountryMap = sc.broadcast(countryMap)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Compute averages and set country names and geoms"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "val averaged = aggregate.mapPartitions{iter =>\n",
+    "    import org.locationtech.geomesa.utils.geotools.Conversions.RichSimpleFeature\n",
+    "\n",
+    "    iter.flatMap{ case (countryName, sf) =>\n",
+    "        if (sf.getType.getTypeName == \"countryInformation\") {\n",
+    "            sf.getProperties.foreach{ prop =>\n",
+    "                val name = prop.getName.toString\n",
+    "                if (name.startsWith(\"total_\")) {\n",
+    "                    val count = sf.get[Integer](\"count\")\n",
+    "                    val avg = (prop.getValue) match {\n",
+    "                            case (a: Integer) => a / count\n",
```

Comment:
Though the reason I can't make is as simple as your example, @cne1x is that `getValue` returns an Object, so I need to pattern match before calling `toDouble`. Also I can't just do `asInstanceOf[Double]` because types like `java.lang.Long` cannot be cast to `Double`